### PR TITLE
Feature/home screen

### PR DESCRIPTION
**Pull request by knowpage**

### CODE REVIEW

File: lib/database/model/movie.dart [hunk 1]
```diff
@@ -1,20 +1,53 @@
...
-  final String? category;
-  final String? movieImg;
+import 'package:json_annotation/json_annotation.dart';
+
+part 'movie.g.dart';
 
+@JsonSerializable()
+class Movie {
+  @JsonKey()
+  String? poster_path;
```
Comment (ntlam6) [inline lib/database/model/movie.dart:8]: use camelCase 

@JsonKey(name: "poster_path);
String? posterPath;

File: lib/database/network/client.dart [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+part 'client.g.dart';
+
+@RestApi(baseUrl: "https://api.themoviedb.org/3/movie/")
+abstract class RestClient {
+  static const String apiKey ='e40e3108780dd1711cbbae5c9f2cfd52';
+  factory RestClient(Dio dio, {String baseUrl}) = _RestClient;
+
+  @GET("popular?api_key=$apiKey&language=en-US&page=1")
+  Future<ResponseData<List<Movie>>> getMovie();
+}
```
Comment (ntlam6) [inline lib/database/network/client.dart:15]: put page -> @Query("page")

File: lib/presentation/journey/home/home_widget.dart [hunk 3]
```diff
@@ -1,67 +1,84 @@
...
+                ),
+                Tab(
+                  text: "Coming Soon",
+                ),
+              ],
+            ),
+          ),
+          body: TabBarView(
+            children: <Widget>[
+              BlocBuilder<HomeBloc, HomeState>(builder: (context, state) {
```
Comment (ntlam6) [inline lib/presentation/journey/home/home_widget.dart:67]: BlocBuilder does nothing here

File: lib/presentation/journey/home/home_widget.dart [hunk 4]
```diff
@@ -1,67 +1,84 @@
...
+                return NowShowingWidget(
+                  controller: controller,
+                  onPageChanged: (int page) {
+                    context.read<HomeBloc>().add(HomeDoneImgEvent(
```
Comment (ntlam6) [inline lib/presentation/journey/home/home_widget.dart:71]: If already push here then use `_homeBloc ` instead

File: lib/presentation/journey/home/home_widget.dart (hunk 3 above)
Comment (knowpage) [reply lib/presentation/journey/home/home_widget.dart:67]: deleted

File: lib/presentation/journey/home/home_widget.dart (hunk 4 above)
Comment (knowpage) [reply lib/presentation/journey/home/home_widget.dart:71]: ok

File: lib/database/network/client.dart (hunk 2 above)
Comment (knowpage) [reply lib/database/network/client.dart:15]: ok

File: lib/database/model/movie.dart (hunk 1 above)
Comment (knowpage) [reply lib/database/model/movie.dart:8]: ok

**Pull request merged**